.github/workflows/beam_PostCommit_Java_Examples_Dataflow.yml

```diff
@@ -0,0 +1,89 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: PostCommit Java Examples Dataflow
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+  - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.comment.body || github.event.sender.login}}'
+  cancel-in-progress: true
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: write
+  checks: write
+  contents: read
+  deployments: read
+  id-token: none
+  issues: write
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_PostCommit_Java_Examples_Dataflow_V2:
+    name: ${{matrix.job_name}} (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 180
+    strategy:
+      matrix:
+        job_name: [beam_PostCommit_Java_Examples_Dataflow_V1]
+        job_phrase: [Run Java examples on Dataflow]
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Java Examples on Dataflow'
+    steps:
+    - uses: actions/checkout@v4
+    - name: Setup repository
+      uses: ./.github/actions/setup-action
+      with:
+        comment_phrase: ${{ matrix.job_phrase }}
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    - name: run PostCommit Java Examples Dataflow script
+      uses: ./.github/actions/gradle-command-self-hosted-action
+      with:
+        gradle-command: :runners:google-cloud-dataflow-java:examples:javaPostCommit
+        max-workers: 12
+    - name: Archive JUnit Test Results
+      uses: actions/upload-artifact@v3
+      if: failure()
+      with:
+        name: JUnit Test Results
+        path: "**/build/reports/tests/"
+    - name: Publish JUnit Test Results
+      uses: EnricoMi/publish-unit-test-result-action@v2
+      if: always()
+      with:
+        files: '**/build/test-results/**/*.xml'
```

Comment:
@Abacn Is it correct to say that we can't see the outcome of this configuration until after the PR is merged?

Reply:
yes. technically it could be tested on fork but need dataflow credential, which is hard to step

This file is copy pasted from beam_PostCommit_Java_Examples_Dataflow_V2, the diff shows below

```
< name: PostCommit Java Examples Dataflow
---
> name: PostCommit Java Examples Dataflow V2
24c24
<   - cron: '0 */6 * * *'
---
>     - cron: '0 */6 * * *'
59,61c59,61
<       matrix:
<         job_name: [beam_PostCommit_Java_Examples_Dataflow_V1]
<         job_phrase: [Run Java examples on Dataflow]
---
>       matrix: 
>         job_name: [beam_PostCommit_Java_Examples_Dataflow_V2]
>         job_phrase: [Run Java Examples on Dataflow Runner V2]
65c65
<       github.event.comment.body == 'Run Java Examples on Dataflow'
---
>       github.event.comment.body == 'Run Java Examples on Dataflow Runner V2'
74c74
<       - name: run PostCommit Java Examples Dataflow script
---
>       - name: run PostCommit Java Examples Dataflow V2 script
77,78c77,78
<           gradle-command: :runners:google-cloud-dataflow-java:examples:javaPostCommit
<           max-workers: 12
---
>             gradle-command: :runners:google-cloud-dataflow-java:examplesJavaRunnerV2IntegrationTest
>             max-workers: 12
```

I was able to run `./gradlew :runners:google-cloud-dataflow-java:examplesJavaRunnerV2IntegrationTest` locally